### PR TITLE
[aggregator] Fix an unintended allocation during element flushing

### PR DESCRIPTION
**Pull request by vdarulis**
**What this PR does / why we need it**:

Looks like it was introduced in a refactor #3901 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/aggregator/aggregator/elem_base.go [hunk 1]
```diff
@@ -216,6 +216,26 @@ type consumeState struct {
...
 
+// Reset resets the consume state for reuse.
+func (c *consumeState) Reset() {
+	var (
+		annotation []byte
+		values     []float64
+	)
+
+	if c.annotation != nil {
+		annotation = c.annotation[:0]
```
Comment (ryanhall07) [inline src/aggregator/aggregator/elem_base.go:227]: am I missing some gc trick here? why can't you just do 

```
c.annotation = c.anntation[:0]
```

File: src/aggregator/aggregator/generic_elem.go [hunk 2]
```diff
@@ -106,14 +106,14 @@ type typeSpecificElemBase interface {
...
 
 type lockedAggregation struct {
-	aggregation typeSpecificAggregation
-	sourcesSeen map[uint32]*bitset.BitSet
-	mtx         sync.Mutex
-	dirty       bool
+	aggregation   typeSpecificAggregation
+	sourcesSeen   map[uint32]*bitset.BitSet
+	mtx           sync.Mutex
+	lastUpdatedAt xtime.UnixNano
```
Comment (ryanhall07) [inline src/aggregator/aggregator/generic_elem.go:112]: this is just to compact the struct to be smaller?

**Review by ryanhall07 — APPROVED** [review]
thanks! good catch!

File: src/aggregator/aggregator/elem_base.go (hunk 1 above)
Comment (vdarulis) [reply src/aggregator/aggregator/elem_base.go:227]: whoops, I assumed doing:
```
	var p []int
	p = p[:0]
```
will panic

File: src/aggregator/aggregator/generic_elem.go (hunk 2 above)
Comment (vdarulis) [reply src/aggregator/aggregator/generic_elem.go:112]: yep, this saves 4-8 bytes of padding

**Pull request merged**